src/media/js/elements/select.js

```diff
@@ -0,0 +1,218 @@
+                        }
+                    }
+                    if (bool) {
+                        this.classList.toggle(VISIBLE, bool);
+                    } else {
+                        this.classList.toggle(VISIBLE);
+                    }
+                    return this;
+                },
+            },
+            select: {
+                // Value is set in the createdCallback.
+                value: null,
+                writable: true
+            }
+        }),
+    });
+
+    function alignMktOptions(mktSelect) {
+        var offset = $('mkt-selected-text', mktSelect).position().left;
+        $('mkt-option', mktSelect).css('padding-left', offset);
+        return offset;
+    }
+
+    function updateMktOptionIndices(mktSelect, selectedOption) {
+        // Sets a data-index on each mkt-option that is not [selected].
+        // Prominently used to set colors since we don't have CSS4 nth-match.
+        var mktOptions = mktSelect.querySelectorAll('mkt-option');
+        var index = 0;
+        for (var i = 0; i < mktOptions.length; i++) {
+            var mktOption = mktOptions[i];
+            if (mktOption === selectedOption) {
+                mktOption.removeAttribute('data-mkt-option--index');
+            } else {
+                mktOption.setAttribute('data-mkt-option--index', index++);
+            }
+        }
+    }
+
+    function copyAttrs(dest, source) {
+        // Copies attributes from one element to another.
+        for (var i = 0; i < source.attributes.length; i++) {
+            var attr = source.attributes[i];
+            if (attr !== 'id') {
+                dest.setAttribute(attr.name, attr.value);
+            }
+        }
+        return dest;
+    }
+
+    function proxyElement(htmlElement, customElement, key) {
+        // Proxy custom element interface to actual element.
+        Object.getOwnPropertyNames(htmlElement.prototype).forEach(function(method) {
+            try {
+                customElement.prototype[method] = {
+                    value: function() {
+                        return this[key][method]();
+                    }
+                };
+            } catch (e) {
+                Object.defineProperty(customElement.prototype, method, {
+                    get: function() {
+                        return this[key][method];
+                    }
+                });
+            }
+        });
+    }
+
+    proxyElement(HTMLSelectElement, MktSelectElement, 'select');
+    proxyElement(HTMLOptGroupElement, MktOptGroupElement, 'optGroup');
+    proxyElement(HTMLOptionElement, MktOptionElement, 'option');
+
+    z.body.on('click', 'mkt-selected', function() {
+        this.parentNode.toggle();
+    })
+
+    .on('change', 'mkt-select', function() {
+        // Update <mkt-selected-text>.
+        var newText = this.querySelector('mkt-option[selected]').innerHTML;
+        this.parentNode.querySelector('mkt-selected-text').innerHTML = newText;
```

Comment:
Doing `parentNode.querySelector` means you can only have one `<mkt-select>` in a container. It might be cleaner to have `<mkt-selected-text for="#the-thing-that-fires-change-events"></mkt-selected-text>` and then `<mkt-selected-text>` setup an event handler in its `createdCallback`.
```js
document.querySelector(this.getAttribute('for'))
        .addEventListener('change', (e) => {
            this.textContent = e.target.selectedOptions[0].textContent;
        });
```

There might be more going on here but you could probably make a generic element for this.